test/fixtures.ts

```diff
@@ -0,0 +1,110 @@
+import type fs from 'fs';
+import type { PathLike } from 'fs';
+import callsite from 'callsite';
+import { DirectoryJSON, fs as memfs, vol } from 'memfs';
+import upath from 'upath';
+
+const realFs = jest.requireActual<typeof fs>('fs');
+
+/**
+ * Class to work with in-memory file-system
+ */
+export class Fixtures {
+  /**
+   * Returns content from fixture file from __fixtures__ folder
+   * @param name name of the fixture file
+   * @param [fixturesRoot] - Where to find the fixtures, uses the current test folder by default
+   * @returns
+   */
+  static get(name: string, fixturesRoot = '.'): string {
+    return realFs.readFileSync(
+      upath.resolve(Fixtures.getPathToFixtures(fixturesRoot), name),
+      {
+        encoding: 'utf-8',
+      }
+    );
+  }
+
+  /**
+   * Adds files from a flat json object to the file-system
+   * @param json flat object
+   * @param cwd is an optional string used to compute absolute file paths, if a file path is given in a relative form
+   */
+  static mock(json: DirectoryJSON, cwd?: string): void {
+    vol.fromJSON(json, cwd);
+  }
+
+  /**
+   * Exports the whole contents of the volume recursively to a flat JSON object
+   * @param paths is an optional argument that specifies one or more paths to be exported. If this argument is omitted, the whole volume is exported. paths can be an array of paths. A path can be a string, Buffer or an URL object.
+   * @param json is an optional object parameter which will be populated with the exported files
+   * @param isRelative is boolean that specifies if returned paths should be relative
+   * @returns
+   */
+  static toJSON(
+    paths?: PathLike | PathLike[],
+    json?: Record<string, unknown>,
+    isRelative?: boolean
+  ): DirectoryJSON {
+    return vol.toJSON(paths, json, isRelative);
+  }
+
+  /**
+   * Removes all files from the volume.
+   */
+  static reset(): void {
+    vol.reset();
+  }
+
+  // Temporary solution, when all tests will be rewritten to Fixtures mocks can be moved into __mocks__ folder
+  static fsExtra(): any {
+    return {
+      ...memfs,
+      pathExists: jest.fn().mockImplementation(pathExists),
+      remove: jest.fn().mockImplementation(memfs.promises.rm),
+      readFile: jest.fn().mockImplementation(readFile),
```

Comment:
```suggestion
      readFile: jest.fn().mockImplementation(memfs.promise.readFile),
```

So we don't need the wrapper function?

Reply:
You are absolutely right! To be honest, I did not remember why I Introduced function with default encoding. Changed